.github/workflows/generate_provenance.yaml

```diff
@@ -0,0 +1,53 @@
+# This workflow wraps the SLSA generator reusable workflow and adds an
+# additional step to simulate a dry run. This allows testing downstream jobs in
+# the caller workflow. A reusable workflow is required, as opposed to a job
+# with multiple steps, because `generator_generic_slsa3` iteslf is a reusable
+# workflow and can only be referenced at the top-level `jobs.*.uses' key. In
+# particular it cannot be used within a step guarded by the if.
+name: Generate Provenance
+
+on:
+  workflow_call:
+    inputs:
+      attestation-name:
+        description:
+          'Custom name for the provenance attestation generated by the workflow.'
+        required: true
+        type: string
+      base64-subject:
+        description:
+          'Base64-encoded string containing the name and sha256 digest of the
+          file to generate the provenance for.'
+        required: true
+        type: string
+
+jobs:
+  # This job calls the generic SLSA workflow to generate provenance.
+  # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
+  generate_provenance:
+    # The `generator_generic_slsa3` workflow fails on `pull_request` events.
```

Comment:
Why though? Is it intentional or a bug? Is it worth following up with them (or have you already)?

Reply:
This is by design so the secrets won't be revealed I think. But I have discussed the need for a dry-run option. Not sure what is the status of that.